### ISSUE TITLE
Closing eshell buffer will also close normal buffer

### ISSUE THREAD

**Issue by ppwwyyxx** [hlissner/doom-emacs#593]
### Steps to reproduce

1. Open a file with Emacs
2. `M-x +eshell:run` to open a shell below. Now two buffers are visible.
3. `exit<RET>` to exit the shell. The shell buffer exits. The other buffer becomes `*scratch*`.

Expected behavior: the other buffer stays

I found that this can be solved by changing:
```elisp
  (+popup-define "^\\*doom \\(?:term\\|eshell\\)"
    '((size . 0.25) (vslot . -10))
    '((select . t) (quit) (transient . 0)))
```
to
```elisp
  (+popup-define "^\\*doom \\(?:term\\|eshell\\)"
    '((size . 0.25) (vslot . -10))
    '((select . t) (quit) (transient . 1)))
```
in `ui/popup/config.el`.

### System information

<details>
<summary>Click to expand</summary>

- OS: gnu/linux (x86_64-pc-linux-gnu)
- Emacs: 27.0.50 (May 10, 2018)
- Doom: 2.0.9 (develop https://github.com/hlissner/doom-emacs/commit/b4a7e5348a6ff2bffacd1e2a9677532a1c515b5e)
- Graphic display: t (daemon: nil)
- System features: XPM JPEG TIFF GIF PNG RSVG SOUND DBUS NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 MODULES THREADS LIBSYSTEMD JSON LCMS2
- Details:
  ```elisp
  uname -a:  Linux KeepLearning 4.16.6-1-ARCH #1 SMP PREEMPT Mon Apr 30 12:30:03 UTC 2018 x86_64 GNU/Linux
  modules:   (:feature debugger eval file-templates lookup services snippets spellcheck (syntax-checker +childframe) version-control workspaces :completion (ivy +childframe) :ui doom-modeline evil-goggles hl-todo nav-flash neotree (popup +defaults) unicode vi-tilde-fringe window-select :tools editorconfig electric-indent eshell imenu make magit password-store pdf rgb rotate-text term tmux :lang cc data emacs-lisp ess go (haskell +intero) (java +meghanada) javascript latex lua markdown (org +attach +babel +capture +export +present +publish) python ruby sh web :collab :config)
  packages:  (ace-link ace-window adaptive-wrap all-the-icons anaconda-mode android-mode anzu async auctex auto-compile auto-yasnippet avy ccls (centered-window :recipe (centered-window :fetcher github :repo anler/centered-window-mode)) cmake-mode coffee-mode command-log-mode company company-anaconda company-auctex company-box company-dict (company-glsl :recipe (company-glsl :fetcher github :repo Kaali/company-glsl)) company-go company-inf-ruby company-irony company-irony-c-headers company-lsp company-lua company-shell company-statistics company-web counsel counsel-css counsel-projectile cuda-mode demangle-mode dired-k disaster dockerfile-mode doom-themes dumb-jump editorconfig emmet-mode eslintd-fix ess ess-R-data-view ess-smart-equals evil evil-anzu evil-args evil-collection evil-commentary evil-easymotion evil-embrace evil-escape evil-exchange evil-goggles evil-indent-plus evil-magit evil-matchit evil-mc evil-multiedit evil-numbers evil-org evil-snipe evil-surround evil-textobj-anyblock evil-vimish-fold evil-visualstar (exec-path-from-shell :ignore t) expand-region flycheck flycheck-cask flycheck-irony flycheck-popup-tip flycheck-posframe flyspell flyspell-correct flyspell-correct-ivy git-gutter-fringe git-link git-timemachine gitconfig-mode gitignore-mode glsl-mode go-eldoc go-guru go-mode gorepl-mode graphql-mode groovy-mode haml-mode haskell-mode haxor-mode helpful hide-mode-line highlight-indentation highlight-numbers highlight-quoted hindent hl-todo htmlize hydra imenu-anywhere imenu-list impatient-mode inf-ruby intero irony irony-eldoc ivy ivy-bibtex ivy-hydra ivy-posframe ivy-rich ivy-rtags ivy-xref js2-mode js2-refactor json-mode kurecolor less-css-mode lsp-mode lsp-ui lua-mode macrostep magit magithub makefile-executor markdown-mode markdown-toc meghanada mips-mode modern-cpp-font-lock moonscript multi-term nasm-mode nav-flash neotree nodejs-repl nose ob-go ob-mongo ob-sql-mode ob-translate opencl-mode (org-bullets :recipe (org-bullets :fetcher github :repo Kaligule/org-bullets)) org-download org-plus-contrib org-tree-slide (osx-clipboard :ignore t) overseer ox-pandoc ox-reveal pass password-store pcre2el pdf-tools persistent-soft persp-mode pip-requirements prodigy projectile pug-mode quelpa quickrun rainbow-delimiters rainbow-mode rake realgud restart-emacs rjsx-mode (rotate-text :recipe (rotate-text :fetcher github :repo debug-ito/rotate-text.el)) rspec-mode rtags ruby-refactor sass-mode shrink-path skewer-mode slim-mode slime smartparens smex solaire-mode stylus-mode swiper tide toc-org toml-mode typescript-mode undo-tree unicode-fonts use-package vi-tilde-fringe vimrc-mode visual-fill-column web-beautify web-mode wgrep which-key xref-js2 yaml-mode yard-mode yasnippet)
  elc dirs:  n/a
  exec-path: (/home/wyx/bin /bin /usr/bin /usr/local/bin /usr/local/sbin /opt/cuda/bin /opt/cxoffice/bin /usr/lib/jvm/default/bin /usr/bin/site_perl /usr/bin/vendor_perl /usr/bin/core_perl /usr/lib/emacs/27.0.50/x86_64-pc-linux-gnu)
  ```
</details>

**Comment by hlissner** [comment]
This _should_ be resolved as of 8807471. Let me know if that is the case.

**Comment by ppwwyyxx** [comment]
Fixed!
